test/e2e/infra/vagrant/playbook/roles/common/tasks/kube.yml

```diff
@@ -1,15 +1,16 @@
 - name: Add an apt signing key for Kubernetes
   apt_key:
-    url: https://packages.cloud.google.com/apt/doc/apt-key.gpg
+    url: https://pkgs.k8s.io/core:/stable:/v1.28/deb/Release.key
```

Comment:
It's not a good idea to bind it with a specific version. You may check this thread:
https://github.com/kubernetes/release/issues/2862#issuecomment-1386805391
We may just need to update the location of the file to `/usr/share/keyrings/cloud.google.gpg`?

Reply:
+1 to @luolanzone for not binding to a specific version. I think we can fetch the latest stable release from https://dl.k8s.io/release/stable.txt and extract the major.minor version.

For signing keys I think we should use keys from pkgs.k8s.io as Google-hosted package repositories are removed on Jan 2024.

Reply:
Are we following this https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/#how-to-migrate and replacing the version as suggested, for both the signing key and apt repository?